### PR TITLE
CalibCalorimetry/HcalPlugins: Change return type of ESProducer to unique_ptr.

### PR DESCRIPTION
**Pull request by gartung**

### CODE REVIEW

**Comment by gartung** [comment]
@Dr15Jones

**Comment by ghellwig** [comment]
please test

File: CalibCalorimetry/HcalPlugins/src/HcalTimeSlewEP.cc [hunk 1]
```diff
@@ -56,10 +56,8 @@ HcalTimeSlewEP::produce(const HcalTimeSlewRecord& iRecord){
   std::vector<edm::ParameterSet> p_TimeSlewM2 = pset_.getParameter<std::vector<edm::ParameterSet>>("timeSlewParametersM2");
   std::vector<edm::ParameterSet> p_TimeSlewM3 = pset_.getParameter<std::vector<edm::ParameterSet>>("timeSlewParametersM3");
 
-  ReturnType hcalTimeSlew( 
-	  new HcalTimeSlew(	  
-	  )
-  );
+  ReturnType hcalTimeSlew =
+	  std::make_unique<HcalTimeSlew>();
```
Comment (ghellwig) [inline CalibCalorimetry/HcalPlugins/src/HcalTimeSlewEP.cc]: it would be aesthetically more pleasing to have this now in one line

**Comment by ghellwig** [comment]
please test

**Comment by ghellwig** [comment]
+1

**Comment by fabiocos** [comment]
+1

**Pull request merged**